mitodl/mit-open
# Carousel Makeover: New tabs and Fixed Width Cards

### What are the relevant tickets?
- Closes https://github.com/mitodl/hq/issues/4360

### Description (What does it do?)
Updates our resource carousels. Main changes:
- Swaps out nuka-carousel for `react-slick`, a react wrapper around [slick-carousel](https://kenwheeler.github.io/slick/). Nuka carousel seems buggy / broken in its new version.
- use customized tab component
- move arrows to top
- uses new carousel cards

### Screenshots (if appropriate):



https://github.com/mitodl/mit-open/assets/9010790/6911e821-fec6-494b-8279-73f5ff0ff8f4

<img width="1835" alt="Screenshot 2024-06-07 at 6 20 22 AM" src="https://github.com/mitodl/mit-open/assets/9010790/bca6a5ae-3c5f-4a71-8922-5fb626ad7da5">
<img width="1826" alt="Screenshot 2024-06-07 at 6 19 32 AM" src="https://github.com/mitodl/mit-open/assets/9010790/69283612-e09f-4338-a818-aac3e39dc842">
<img width="440" alt="Screenshot 2024-06-07 at 6 20 33 AM" src="https://github.com/mitodl/mit-open/assets/9010790/3e7799ca-5ed4-4497-af0f-53e36e27b903">
<img width="552" alt="Screenshot 2024-06-07 at 6 19 49 AM" src="https://github.com/mitodl/mit-open/assets/9010790/8a6e2bc7-2816-4e4a-95f9-8e9da168aebd">


### How can this be tested?


### How should this be tested
1. Play with the carousel on the homepage. Do things like:
    - Switch tabs
    - Change pages using buttons
    - change pages by dragging
2. Also, try resize the screen. In particular:
    - **page size:** If 2 cards are fully visible and a third is partially visible, the carousel should move by 2 cards when changing pages. If 3 cards are fully visible and a fourth is partially visible, the carousel should move by 3 cards when srolling, etc.
    - **Button Disabling**: Buttons should be appropriately disabled when there is no next/prev page.
         - For example, if you resize the screen to a pagesize of 2, and scroll to the penultimate page, then resize to 4 cards per page, the Next button should be disabled.
